HIP/hip-260.md

```diff
@@ -122,6 +126,63 @@ final updated value of the storage slot after the completion of the smart
 contract call. Transient states between the start and finish of the contract are
 not stored in the record stream.
 
+### Record Stream File Changes
+
+Currently Hedera nodes produce record stream files that contain a series of
+transactions within a two second interval. The record stream file (.rcd) 
+contains a transaction record for each transaction in that file. For each 
+record stream file there is a corresponding signature file (rcd_sig) 
+that includes the signature generated by the node.
+
+#### Side car records file
+
+A new file of type (.rcd_side) will be introduced to track all side car records
+produced during the same 2 second period as the main record file (.rcd). All
+side car records in the .rcd_side file should have a corresponding main record
+with the same id in the main record file (.rcd).
+
+The sidecar + record file + signature file will all be zipped together into a
+single zip and uploaded to the AWS/Google buckets.
+
+The side car file will have the following structure:
+
+| Name                              | Type (Bytes) |  Description                                                     |
+| --------------------------------- | -----------: | ---------------------------------------------------------------: |
+| Record Stream File Format Version |    int (4)   | "Record stream file format version, Value: 5"                    |
+| Hapi Proto Major Version          |    int (4)   | Matches NetworkGetVersionInfo.hapiProtoVersion Major.Minor.Patch |
+| Hapi Proto Minor Version          |    int (4)   | Major version: 0;                                                |
+| Hapi Proto Patch Version          |    int (4)   | Minor version: 9;                                                |
+| Object stream version             |    int (4)   | Patch version: 0                                                 |
+| 1st RecordStreamObject            |    byte[]    | Serialized RecordStreamObject bytes.                             |
+| 2nd RecordStreamObject            |    byte[]    | ...                                                              |
+| ...                               |    byte[]    | ...                                                              |
+
+#### Record file names
+
+Side car records will be externalised in a new file with the same name as 
+the main record stream file and the new extension ".rcd_side", e.g.
+
+Record File         (.rcd): 2020-10-19T21_35_39.000000000Z.rcd
+Side car File  (.rcd_side): 2020-10-19T21_35_39.000000000Z.rcd_side
+
+#### Record file structure update
+
+Record files are to be updated with a new property appended at the end of the 
+version 5 record files. The property will be encoded after the long blockNumber
```

Comment:
> I don't believe adding a field at the end of a v5 file is backwards compatible. So we should create a v6 record file.

Richard wanted to bundle the updates for the block HIP and sidecar HIP together and have a single update to the record file, that's why I proposed using v5.
> Richard had said he wanted to use protobuf for the new file.

Not sure what that implies.
> We should also keep in mind the block concept HIP that will be creating a new version and coordinate the two.

AFAIK the block concept HIP adds the `blockNumber`, that's why I specified here that the `SideCarHash` will be encoded after the `blocknumber` property.

Reply:
I'm saying that adding fields to the end of the v5 format would necessitate a v6 format since it's not backwards compatible to clients. And if we're doing a new version anyway we should upgrade the format from its current complex, proprietary binary format to our usual standard of using protobuf.